Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/ContestsPage.tsx

```diff
@@ -31,30 +33,52 @@ const ContestsPage = () => {
         [],
     );
 
-    const handlePageChange = (page: number) => {
-        setPage(page);
-    };
-
-    useEffect(
-        () => {
-            setSearchParams(filters.reduce((p:any, { type, value }) => {
-                const values = isNil(p[type])
-                    ? []
-                    : p[type];
-                values.push(value);
-
-                return {
-                    ...p,
-                    [type]: values,
-                };
-            }, {}));
+    const handlePageChange = useCallback(
+        (page: number) => {
+            applyFilters(filters, page);
         },
-        [ filters, setSearchParams ],
+        [ applyFilters, filters ],
     );
 
+    const handleFilterClick = useCallback((filter: IFilter) => {
+        const { name: filterName, type, value } = filter;
+        const paramName = type.toString();
+        searchParams.delete(paramName);
+        searchParams.delete(paramName.toLowerCase());
+
+        const removeFilter = filters.some(({ name }) => name === filterName) &&
```

Comment:
`removeFilter` implies this is an action, i.e. a function. `shouldRemoveFilter` is better